From ca8cf3515dcbde229702b4caf16e267c1708f64f Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 2 Jul 2015 15:14:38 -0400
Subject: [PATCH 0515/1936] Fix Midi Learn In Effects

This commit also fixes some other controls such as the subsynth
Phmag* controls
---
 rtosc                     |  2 +-
 src/Effects/EffectMgr.cpp | 13 +++++++++----
 src/UI/SUBnoteUI.fl       |  3 ---
 3 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/rtosc b/rtosc
index 9c73162f..cdd23927 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 9c73162fe05ca82a60c49bd24e41b097d004703d
+Subproject commit cdd23927f541db537ab04fb9c533cdc7434f16bc
diff --git a/src/Effects/EffectMgr.cpp b/src/Effects/EffectMgr.cpp
index 3ad54fa2..1a96f9ef 100644
--- a/src/Effects/EffectMgr.cpp
+++ b/src/Effects/EffectMgr.cpp
@@ -52,8 +52,10 @@ static const rtosc::Ports local_ports = {
 
             if(!rtosc_narguments(msg))
                 d.reply(d.loc, "i", eff->geteffectparrt(atoi(mm)));
-            else 
+            else {
                 eff->seteffectparrt(atoi(mm), rtosc_argument(msg, 0).i);
+                d.broadcast(d.loc, "i", eff->geteffectparrt(atoi(mm)));
+            }
         }},
     {"preset::i", rProp(alias) rDoc("Effect Preset Selector"), NULL,
         [](const char *msg, rtosc::RtData &d)
@@ -61,8 +63,10 @@ static const rtosc::Ports local_ports = {
             EffectMgr *eff = (EffectMgr*)d.obj;
             if(!rtosc_narguments(msg))
                 d.reply(d.loc, "i", eff->getpreset());
-            else
+            else {
                 eff->changepresetrt(rtosc_argument(msg, 0).i);
+                d.broadcast(d.loc, "i", eff->getpreset());
+            }
         }},
     {"eq-coeffs:", rProp(internal) rDoc("Get equalizer Coefficients"), NULL,
         [](const char *, rtosc::RtData &d)
@@ -81,9 +85,10 @@ static const rtosc::Ports local_ports = {
     {"efftype::i", rDoc("Get Effect Type"), NULL, [](const char *m, rtosc::RtData &d)
         {
             EffectMgr *eff  = (EffectMgr*)d.obj;
-            if(rtosc_narguments(m)) 
+            if(rtosc_narguments(m))  {
                 eff->changeeffectrt(rtosc_argument(m,0).i);
-            else
+                d.broadcast(d.loc, "i", eff->nefx);
+            } else
                 d.reply(d.loc, "i", eff->nefx);
         }},
     {"efftype:b", rProp(internal) rDoc("Pointer swap EffectMgr"), NULL,
diff --git a/src/UI/SUBnoteUI.fl b/src/UI/SUBnoteUI.fl
index 961ef572..40d397f6 100644
--- a/src/UI/SUBnoteUI.fl
+++ b/src/UI/SUBnoteUI.fl
@@ -58,7 +58,6 @@ class PPSlider {: {public Fl_Slider, public Fl_Osc_Widget}
 	bool ctl_click    = (event == FL_PUSH && Fl::event_state(FL_BUTTON1) && Fl::event_ctrl());
 	bool shift_middle = (event == FL_PUSH && Fl::event_state(FL_BUTTON2) && Fl::event_shift());
 	if(middle_mouse || ctl_click) {
-		puts("Trying to learn...");
 		osc->write("/learn", "s", (loc+ext).c_str());
 		return 1;
 	} else if(shift_middle) {
@@ -153,9 +152,7 @@ if (Fl::event_button1() || Fl::event() == FL_MOUSEWHEEL) x=127-(int)o->value();
 make_window();
 harmonic->show();
 mag->ext = "Phmag" + to_s(n);
-puts((mag->loc + mag->ext).c_str());
 mag->oscRegister(mag->ext.c_str());
-puts((base + "Phrelbw"+to_s(n)).c_str());
 osc->createLink(base+"Phrelbw"+to_s(n), bw);
 osc->requestValue(base+"Phrelbw"+to_s(n));
 
-- 
2.47.0

